include/task_manager.h

```diff
@@ -77,11 +76,6 @@ namespace detail {
 
 			auto unique_tsk = into_task(std::move(cgh));
 
-			// Require the collective group before inserting the task into the ring buffer, otherwise the executor will try to schedule the collective host
-			// task on a collective-group thread that does not yet exist.
-			// The queue pointer will be null in non-runtime tests.
-			if(m_queue) m_queue->require_collective_group(unique_tsk->get_collective_group_id());
```

Comment:
I guess this issue doesn't happen in instruction graph execution?

Reply:
The IDAG RT does not tie collective host tasks to particular threads, so this is not relevant anymore.

But I also think the comment was somewhat misleading, since `host_queue` could always just have spawned new CG threads as-needed. I believe the original reason was that doing this from the TM enforces that all `MPI_Comm_dup`s happen in a single global total order. This is now guaranteed through IDAG dependencies on the `clone_collective_group_instruction` we emit for the first use of each new CG.